python/ray/data/_internal/logical/optimizers.py

```diff
@@ -1,10 +1,21 @@
-from typing import List
+from typing import List, Iterator
 
+from ray.data.block import Block
+from ray.data._internal.compute import is_task_compute, CallableClass, get_compute
```

Comment:
Seems we have to depend on `ray.data._internal.compute` here. Let's add a TODO to refactor those needed methods out of `ray.data._internal.compute`. So we don't take a dependency on it in the future - i.e. we plan to delete `compute.py` eventually.

Reply:
Yep I decided to accept this dependency for now and thought that we could remove this dependency when we delete `compute.py`, otherwise we start to accumulate repeated definitions and run the risk of drift. Some of these like `CallableClass` should be moved to a user-facing interfaces file (since the type is user-facing), while `is_task_compute()` and `get_compute()` could probably be moved into this module, but I held off on that for now to try to keep this PR size from getting too large.